backend/tests/unit/dal/Services/ProjectServiceTest.cs

```diff
@@ -2097,6 +2097,234 @@ public async void SetStatusAsync_NotInSpl_InvalidOperationException()
+            project.StatusId = contractInPlace.Id; // Contract in Place status
+
+            EntityHelper.CreateAgency(2);
+            parcel.AgencyId = 2;
+            EntityHelper.CreatePropertyClassification(2, "new classification");
+            parcel.ClassificationId = 2;
+            project.Properties.First().Parcel = parcel;
+
+            // Act
+            var result = await service.SetStatusAsync(project, project.Workflow.Code);
+
+            // Assert
+            Assert.NotNull(result);
+            result.StatusId.Should().Be(contractInPlace.Id);
+            result.Status.Should().Be(contractInPlace);
+            var property = result.Properties.First().Parcel;
+            property.AgencyId.Should().Be(2);
+            property.ClassificationId.Should().Be(2);
+            property.IsVisibleToOtherAgencies.Should().BeFalse();
+        }
+
+        [Fact]
+        public async void SetStatusAsync_Disposed_Success()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.DisposeApprove, Permissions.AdminProjects, Permissions.ProjectView).AddAgency(1);
+
+            var init = helper.InitializeDatabase(user);
+            var workflows = init.CreateDefaultWorkflowsWithStatus();
+            init.SaveChanges();
+            var project = init.CreateProject(1, 1);
+            init.SetStatus(project, "SPL", "DIS");
+            var parcel = init.CreateParcel(1);
+            project.AddProperty(parcel);
+            parcel.ProjectNumber = project.ProjectNumber;
+            project.DisposedOn = DateTime.Now; // required for Marketed.
+            init.SaveChanges();
+
+            var options = ControllerHelper.CreateDefaultPimsOptions();
+            var service = helper.CreateService<ProjectService>(user, options);
+
+            var disposed = init.ProjectStatus.First(s => s.Code == "DIS");
+            project.StatusId = disposed.Id; // Marketed status
+
+            EntityHelper.CreateAgency(2);
+            parcel.AgencyId = 2;
+            EntityHelper.CreatePropertyClassification(2, "new classification");
+            parcel.ClassificationId = 2;
+            project.Properties.First().Parcel = parcel;
+
+            // Act
+            var result = await service.SetStatusAsync(project, project.Workflow.Code);
+
+            // Assert
+            Assert.NotNull(result);
+            result.StatusId.Should().Be(disposed.Id);
+            result.Status.Should().Be(disposed);
+            result.DisposedOn.Should().NotBeNull();
+            var property = result.Properties.First().Parcel;
+            property.AgencyId.Should().Be(2);
+            property.ClassificationId.Should().Be(2);
+            property.IsVisibleToOtherAgencies.Should().BeFalse();
+        }
+
+        [Fact]
+        public async void SetStatusAsync_Disposed_InvalidOperationException()
```

Comment:
This will need some more tests when the backend logic for disposed is finished (disposed classification, null agency).